ui/nuxeo-routing-behavior.js

```diff
@@ -98,19 +102,66 @@ export const RoutingBehavior = {
   _computeNavigateTo() {
     return function(...args) {
       if (this.router) {
-        const route = args[0];
+        const [route, ...params] = args;
         const baseUrl = this.router.baseUrl || '';
-        if (route.startsWith('/')) {
-          this.router.navigate(baseUrl + route);
+        let path;
+        if (typeof route === 'object') {
+          path = this._routeEntity(...args);
+        } else {
+          if (route.startsWith('/')) {
+            this.router.navigate(baseUrl + route);
+          }
+          if (!this.router[route]) {
+            console.error(`Could not navigate to a url for route ${route}`);
+          }
+          path = this.router[route].apply(this, params);
         }
-        if (!this.router[route]) {
-          console.error(`Could not navigate to a url for route ${route}`);
-        }
-        const params = Array.prototype.slice.call(args, 1);
-        this.router.navigate(this.router[route].apply(this, params));
+        this.router.navigate(path);
       } else {
         console.error('No router defined');
       }
     };
   },
+
+  _routeEntity(...args) {
+    if (args.length === 0) {
+      return;
+    }
+    const [obj, ...params] = args;
+    if (typeof obj !== 'object') {
+      throw new Error(`cannot resolve route: "${obj}" is not a valid entity object`);
+    }
+    let entityType = obj['entity-type'];
+    if (!entityType) {
+      // XXX Sometimes we don't have the entity type. For example, on nuxeo-document-storage, we were not storing it.
+      // In such cases, we'll assume we are dealing with a document if it has both a path and a uid properties.
+      if (obj.path && obj.uid) {
+        entityType = 'document';
+      } else {
+        throw new Error(`cannot resolve route: object does not have an "entity-type"`);
+      }
+    }
+    let routeKey =
+      Nuxeo &&
+      Nuxeo.UI &&
+      Nuxeo.UI.config &&
+      Nuxeo.UI.config.router &&
+      Nuxeo.UI.config.router.key &&
+      Nuxeo.UI.config.router.key[entityType];
```

Comment:
shouldn't we have a default key for other entities too? (ie `id`) ?

Reply:
We could! How would we name such a config property? `Nuxeo.UI.config.router.defaultKey`, just in case someone wants to have and entity type named default?

Reply:
Goal would just be to have a hardcoded default like `id`, no need to have another conf property to set it imho